spec/rubocop/cop/lint/number_conversion_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::NumberConversion do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense when using `#to_i`' do
+    expect_offense(<<-RUBY.strip_indent)
+      10.to_i
+      ^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using 10.to_i, use stricter Integer(10).
```

Comment:
This is the opposite 😅 
I was concerned about the test case for `String#to_i`.

```diff
-10.to_i
+'10'.to_i
```

https://github.com/bbatsov/rubocop/pull/5147#discussion_r159858342

Reply:
😅 I've changed it to string, also added specs to cover *10.to_i* scenario as well.